### PR TITLE
D3ASIM-2928: Moving simulation_calculation bases classes from d3a to …

### PR DESCRIPTION
**Pull request by faizan2590**
…d3a-interface.

### CODE REVIEW

File: setup.py
Comment (hannesdiedrich) [inline setup.py:10]: please revert before merging

File: src/d3a/models/strategy/area_agents/one_sided_alternative_pricing_agent.py
Comment (hannesdiedrich) [inline src/d3a/models/strategy/area_agents/one_sided_alternative_pricing_agent.py:41]: How about adding a `is_house_node` into the utily.py? You always strike me as you prefer a function than a logic  chain. :-)

**Review by hannesdiedrich — DISMISSED** [review]
One comment, once addressed, LGTM

File: src/d3a/models/strategy/area_agents/one_sided_alternative_pricing_agent.py
Comment (faizan2590) [reply src/d3a/models/strategy/area_agents/one_sided_alternative_pricing_agent.py:41]: Indeed I prefer a function than a logic chain to avoid code repetition but that was the only usage of `_is_house_node `.

**Review by spyrostz — DISMISSED** [review]
LGTM

File: src/d3a/d3a_core/sim_results/endpoint_buffer.py
Comment (hannesdiedrich) [inline src/d3a/d3a_core/sim_results/endpoint_buffer.py:63]: Just for my understanding: We need kpi and market_bills for the external Strategies (market_info and market_stats) right?

**Review by hannesdiedrich — DISMISSED** [review]
One clarification, LGTM

File: src/d3a/d3a_core/sim_results/endpoint_buffer.py
Comment (faizan2590) [reply src/d3a/d3a_core/sim_results/endpoint_buffer.py:63]: Exactly, ideally it could be done on d3a-web too and when the external user needs it, it could be read from DB and sent via d3a-web that way we can further optimize d3a and release resources from d3a to even calculate these results

**Review by hannesdiedrich — APPROVED** [review]
LGTM, please don't forget https://github.com/gridsingularity/d3a-interface/pull/137#discussion_r525923077

File: src/d3a/d3a_core/d3a_jobs.py
Comment (spyrostz) [inline src/d3a/d3a_core/d3a_jobs.py:46]: Please revert before merging

File: src/d3a/d3a_core/redis_connections/redis_communication.py
Comment (spyrostz) [inline src/d3a/d3a_core/redis_connections/redis_communication.py:246]: This will crash on the d3a-web/results service side. Better to revert to the old behavior

File: src/d3a/d3a_core/simulation.py
Comment (spyrostz) [inline src/d3a/d3a_core/simulation.py:68]: I think this is not needed here.

**Review by spyrostz — APPROVED** [review]
Some small comments, LGTM once resolved.

**Pull request merged**
